Assets/Plugins/iOS/Shopify/Checkout/Checkout.WebViewController.swift

```diff
@@ -56,28 +57,73 @@ class WebViewController: UIViewController {
     
     override func loadView() {
         super.loadView()
-        self.view = WebContentView(frame: view.frame)
+        self.view = WKWebView(frame: view.frame)
     }
     
     override func viewDidLoad() {
         super.viewDidLoad()
         webView.navigationDelegate = self
-        contentView.setActionButton(target: self, selector: #selector(didPressCancel))
-        
+        setupNavigationBar()
         loadCheckoutURL()
     }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        webView.addObserver(self, forKeyPath: #keyPath(WKWebView.estimatedProgress), options: .new, context: &kvoContext)
```

Comment:
I would put observer registration and unregistration in `init` and `dealloc` respectively. Having it in `viewWillAppear` and `viewWillDisappear` can cause weird behaviour when this view can still be visible after it "disappears".